Documentation/Class-Injection.md

```diff
@@ -79,25 +79,81 @@ var someInstance = new MyClass(pointer);
-  it's referenced from managed domain. Attempting to use that object afterwards will result
-  in `ObjectCollectedException`. Conversely, managed representation of injected object will not be garbage collected as
-  long as it's referenced from IL2CPP domain.
-* It might be possible to create a cross-domain reference loop that will prevent objects from being garbage collected.
-  Avoid doing anything that will result in injected class instances (indirectly) storing references to itself. The
-  simplest example of how to leak memory is this:
+## Garbage Collection
+
+* Managed instances of injected classes hold strong handles to their unmanaged counterparts.
+  This means it is unlikely for an unmanaged object to be collected while it is referenced from the managed domain.
+  If this rare case does occur, an attempt to use to injected instance will throw an `ObjectCollectedException`.
+* If there are no extant references to the managed instance, it will be garbage collected,
+  releasing the strong handle it holds and allowing the underlying unmanaged object to eventually be collected in turn.
+  When this occurs, however, the finalizer for the managed instance will not be run,
+  delaying execution until the unmanaged object is also ready to be garbage collected.
+* The unmanaged-to-managed mapping is a one-to-many relationship.
+  While during typical execution there will be exactly zero, or exactly one extant managed object,
+  in certain situations, such as when the object pool is disabled, there may be any number.#
+* Finalizers are supported for injected classes, but note that the managed injector (`~MyClass`) is not invoked predictably.
+  If you want to implement a custom finalizer, define a `private void Il2CppFinalize()` method on your class (see below).
+* Due to the implementation of finalizers, the "resurrection pattern" does not translate directly to injected types.
+  If you do not call `ClassInjector.ResurrectObject` during an object's finalizer,
+  the unmanaged object will be garbage collected and future use will throw `ObjectCollectedException`s.
+  **NB:** You will still also need to call `Il2CppSystem.GC.ReRegisterForFinalize` if applicable.
+
+<details>
+<summary>A detailed example of finalizer behavior for injected classes</summary>
+<br>
+
+Consider the following example:
 
 ```c#
-class Injected: Il2CppSystem.Object {
-    Il2CppSystem.Collections.Generic.List<Il2CppSystem.Object> list = new ...;
-    public Injected() {
-        list.Add(this); // reference to itself through an IL2CPP list. This will prevent both this and list from being garbage collected, ever.
+class Foo : Il2CppSystem.Object
+{
+    public Foo(IntPtr ptr) : base(ptr) { }
+    public Foo() : this(ClassInjector.DerivedConstructorPointer<Foo>())
+    {
+        ClassInjector.DerivedConstructorBody(this);
+    }
+
+    private void Il2CppFinalize()
```

Comment:
v2 replaces `[HideFromIl2Cpp]` with an opt-in system. The user indicator for this would be:
```cs
[Il2CppMethod(Name = "Finalize")]
```

Reply:
The Name property applies in any situation where an Il2Cpp name differs from the generated name in managed code. In the case of finalizers, the v2 generator appends underscores, which is approximately the same as what you're doing.